src/main.rs

```diff
@@ -50,10 +50,64 @@ fn main() {
     }
 }
 
+fn build_cache_without_training(cl: clap::ArgMatches) -> Result<(), Box<dyn Error>> {
```

Comment:
Given that we have `cache.rs`, perhaps consider moving this logic there (so we have cache-related methods all in one place?

Reply:
im a bit hesitant since most of the lines are more connected to the buffer and arguments of cmd

Reply:
fair point